src/coreComponents/fileIO/vtk/VTKPolyDataWriterInterface.cpp

```diff
@@ -332,21 +373,154 @@ void VTKPolyDataWriterInterface::writeElementFields( vtkSmartPointer< vtkCellDat
+    //   For now, we apply a special component-by-component treatment to 4-dimension fields (see below), but we do not output fields with 5
+    // dimensions or more.
+
+    // First, check if the field requires a special treatment. For now, array4d< real64 > (for instance, for phaseCompFraction) are
+    // treated separately to make sure that they can be easily visualized in Paraview.
+    int numArrayDims = 0;
+
     er.forElementSubRegions< SUBREGION >( [&]( auto const & esr )
     {
       WrapperBase const & wrapper = esr.getWrapperBase( field );
-      writeField( wrapper, data, esr.size(), count );
+      numArrayDims = wrapper.numArrayDims();
     } );
-    celldata->AddArray( data );
+
+    // Then, output the field to VTK
+    // Note: fields with 5 dimensions or more are not supported and are not output to file below
+    if( numArrayDims < 4 )
+    {
+      // If the field has three dimensions or less, it can be output in a generic fashion
+      writeElementField1D2D3D< SUBREGION >( celldata, er, field );
+    }
+    else if( numArrayDims == 4 )
+    {
+      writeElementField4D< SUBREGION >( celldata, er, field );
+    }
+    else
+    {
+      GEOSX_ERROR( "Fields with more than 4 dimensions cannot be output to VTK" );
+    }
   }
 }
+
+template< class SUBREGION >
+void VTKPolyDataWriterInterface::writeElementField1D2D3D( vtkSmartPointer< vtkCellData > const celldata,
+                                                          ElementRegionBase const & er,
+                                                          string const & field ) const
+{
+  localIndex count = 0;
+
+  vtkSmartPointer< VTKGEOSXData > data = VTKGEOSXData::New();
+  data->SetNumberOfValues( er.getNumberOfElements< SUBREGION >() );
+  data->SetName( field.c_str() );
+
+  er.forElementSubRegions< SUBREGION >( [&]( auto const & esr )
+  {
+    WrapperBase const & wrapper = esr.getWrapperBase( field );
+    writeField( wrapper, data, esr.size(), count );
+  } );
+  celldata->AddArray( data );
+}
+
+template< class SUBREGION >
+void VTKPolyDataWriterInterface::writeElementField4D( vtkSmartPointer< vtkCellData > const celldata,
+                                                      ElementRegionBase const & er,
+                                                      string const & field ) const
+{
+  // Step 1: we need to figure out the size of the two dimensions of the array that we are going to output component by component
+  localIndex nCompInSecondDim = 0;
+  localIndex nCompInThirdDim = 0;
+  er.forElementSubRegions< SUBREGION >( [&]( auto const & esr )
+  {
+    WrapperBase const & wrapperBase = esr.getWrapperBase( field );
+    std::type_info const & typeID = wrapperBase.getTypeId();
+    rtTypes::applyArrayTypeLambda2( rtTypes::typeID( typeID ),
+                                    true,
+                                    [&]( auto array, auto GEOSX_UNUSED_PARAM( Type ) )
+    {
+      typedef decltype( array ) arrayType;
```

Comment:
```suggestion
      using arrayType = decltype( array );
```
(sorry to nitpick, we probably still have `typedef` in a bunch of places, but I'm trying to root them out)

Reply:
Done, I also removed the `typedef` from the original function `writeField`.